src/Symfony/Component/Messenger/Command/AbstractFailedMessagesCommand.php

```diff
@@ -29,20 +30,25 @@
  */
 abstract class AbstractFailedMessagesCommand extends Command
 {
-    private $receiverName;
-    private $receiver;
+    private $globalFailureReceiverName;
+    private $failureTransports;
 
-    public function __construct(string $receiverName, ReceiverInterface $receiver)
+    public function __construct(?string $globalFailureReceiverName, $failureTransports)
     {
-        $this->receiverName = $receiverName;
-        $this->receiver = $receiver;
+        $this->failureTransports = $failureTransports;
+        $this->globalFailureReceiverName = $globalFailureReceiverName;
+        if (!$failureTransports instanceof ServiceLocator) {
```

Comment:
We should probably move this up ONE line so it's right after `$this->failureTransports`.

Reply:
but if we do that it's replaced afterwards. Better to put like this to avoid the else right? Am I missing something?